### DISCUSSION TITLE
1.28.0

### DISCUSSION THREAD

**Discussion by Koenkk** [Koenkk/zigbee2mqtt#14269] [General]
# Notes
- For those using a CC1352 or CC2652 based coordinator and are running larger networks (80+ devices): a new beta firmware is available which should improve reliability and performance. The firmware can be downloaded from [here](https://github.com/Koenkk/Z-Stack-firmware/tree/develop/coordinator/Z-Stack_3.x.0/bin), leave your feedback [here](https://github.com/Koenkk/Z-Stack-firmware/issues/383).

# Features
- #13780  Support install codes (frontend: `Settings -> Tools -> Add install code`)
 
# Improvements
- [#573](https://github.com/Koenkk/zigbee-herdsman/pull/573) Support joining devices via specific routers for EZSP adapter
- [#4583](https://github.com/Koenkk/zigbee-herdsman-converters/pull/4583) Add the possibility to activate LED when the device is on for Legrand 067775/741811 and rename `led_when_off`  to `led_in_dark` and `led_when_on` to `led_if_on`
- #13861 Allow read of `illuminance`/`illuminance_lux` value for Xiaomi GZCGQ01LM and KMPCIL KMPCIL_RES005
- [#4630](https://github.com/Koenkk/zigbee-herdsman-converters/pull/4630) Add `power_on_behavior` to Legrand  064882, 067771, 067775/741811 and `led_in_dark` to 067775/741811
- [#4638](https://github.com/Koenkk/zigbee-herdsman-converters/pull/4638) Add pincode management to Kiwkset 99140-031
- [#4639](https://github.com/Koenkk/zigbee-herdsman-converters/pull/4639) Support OTA for IKEA E2007
- #13927 Support power measurements for Sengled E12-N1E
- [#4647](https://github.com/Koenkk/zigbee-herdsman-converters/pull/4647) Add battery reporting for Sengled E1D-G73WNA
- [#4649](https://github.com/Koenkk/zigbee-herdsman-converters/pull/4649) Add eco and running mode for Owon PCT504
- #13965 Log MQTT disconnect reason
- [#4669](https://github.com/Koenkk/zigbee-herdsman-converters/pull/4669) Add OTA support for Xiaomi WS-EUK03
- [#4674](https://github.com/Koenkk/zigbee-herdsman-converters/pull/4674) Expose battery for Xiaomi SRTS-A01
- [#4668](https://github.com/Koenkk/zigbee-herdsman-converters/pull/4668) Add more functionality to ShinaSystem CSM-300ZB_V2 and DMS-300ZB
- [#4676](https://github.com/Koenkk/zigbee-herdsman-converters/pull/4676) Expose energy for LifeControl MCLH-03
- [#4678](https://github.com/Koenkk/zigbee-herdsman-converters/pull/4678) Support energy metering for ROBB ROB_200-014-0
- #14103 Improve Home Assistant history view for sensors without unit
- #13464 Support display_unit for Xiaomi VOCKQJK11LM
- [#4688](https://github.com/Koenkk/zigbee-herdsman-converters/pull/4688) Support `operation_mode` and `rf_pairing` for ShinaSystem ISM300Z3 
- [#4691](https://github.com/Koenkk/zigbee-herdsman-converters/pull/4691) Support second channel for HZC Electric D086-ZG
- [#4692](https://github.com/Koenkk/zigbee-herdsman-converters/pull/4692) Dynamic Siglis ZFU-1D-CH exposes 
- [#4713](https://github.com/Koenkk/zigbee-herdsman-converters/pull/4713/) Support battery % for Develco MOSZB-140
- #14200 Fix Home Assistant discovery cover state publishing of multi-endpoint devices

# Fixes
- #13964 #13964 EZSP pairing and stability improvements
- #13813 Fix cover state incorrect for TuYa TS130F
- #13800 Fix TuYa TS0601_dimmer control for `_TZE200_ip2akl4w` and `_TZE200_1agwnems`
- [#4596](https://github.com/Koenkk/zigbee-herdsman-converters/pull/4596) Fix brightness changed to 1 when light is turned on 
- #13843 Fix controlling individual switch of LELLKI WP30-EU
- [#4603](https://github.com/Koenkk/zigbee-herdsman-converters/issues/4603) Fix individual controlling of TS0004
- [#4608](https://github.com/Koenkk/zigbee-herdsman-converters/issues/4608) Expose voltage for Xiaomi QBKG26LM
- #8710 Fix battery % multiplied by 2 for Yale YMF30
- #13812 Fix contact inverted for TS0601_vibration_sensor
- [#4616](https://github.com/Koenkk/zigbee-herdsman-converters/pull/4616) Fix negative power values of Develco EMIZB-132
- [#4614](https://github.com/Koenkk/zigbee-herdsman-converters/issues/4614) Fix TS0201 (`_TZ3000_ywagc4rj`) humidity value incorrect
- #13878  Fix Smartwings WM25L-Z state inverted
- #7807 Fix setting devices/groups to file via environment variables not working
- #13876 Fix Eco-Dim.07/Eco-Dim.10 not detected as supported
- [#4621]( https://github.com/Koenkk/zigbee-herdsman-converters/issues/4621) Fix Enocean PTM 216Z missing command error
- #13919 Expose battery for TuYa TS0202_1
- [#4642](https://github.com/Koenkk/zigbee-herdsman-converters/pull/4642) Fix Siglis ZFP-1A-CH cover state
- #13954 Fix TuYa TS0002_switch_module channels not individually controllable
- #13944 Fix cover state in Home Assistant for multi channel cover controllers
- #13966  Fix Gewiss GWA1531 state inverted
- #13329 Fix incorrect power values for Develco SPLZB-131
- #14023 Fix `motor_state ` value template error in Home Assistant for Xiaomi ZNCLBL01LM
- #13598 Fix battery reporting for TuYa KCTW1Z
- [#4682](https://github.com/Koenkk/zigbee-herdsman-converters/pull/4682) Fix `holiday_start_stop` for TuYa TV02-Zigbee
- [#4687](https://github.com/Koenkk/zigbee-herdsman-converters/pull/4687) Fix `auto_mode` for TuYa TV02-Zigbee
- #14107 Fix LELLKI WP30-EU states incorrect
- [#4703](https://github.com/Koenkk/zigbee-herdsman-converters/pull/4703) Fix time sync for RTX ZVG1
- [#4709](https://github.com/Koenkk/zigbee-herdsman-converters/pull/4709) Fix "broadcast" state readout on Siglis ZFP-1A-CH
- [#4714](https://github.com/Koenkk/zigbee-herdsman-converters/pull/4714) Fix weekdays for timers for RTX ZVG1
- #14193 Fix Moes MS-108ZR cover state
- [#4721](https://github.com/Koenkk/zigbee-herdsman-converters/pull/4721) Fix no battery value for Lidl HG08164
- #14253 Fix power source for Xiaomi QBKG11LM and QBKG12LM
- #14075 Fix crash when checking for update and device does not respond

# New supported devices
This release adds support for **53** new devices:
- `07743L` Immax Neo Smart LED E27 11W RGB + CCT, color, dimmable, Zigbee 3.0
- `1741930V7` Philips Hue Calla outdoor
- `303-0136` HFH Solutions LED controller
- `404065` Müller Licht tint LED-Globe Retro Gold XXL E27
- `4058075729063` LEDVANCE SMART+ Classic B40 E14 dimmable white
- `4058075729100` LEDVANCE SMART+ Classic P40 E14 dimmable white
- `4058075729148` LEDVANCE SMART+ Spot PAR16 50 GU10 dimmable white
- `4058075729209` LEDVANCE SMART+ Filament Classic A 52 E27 Amber dimmable
- `4058075729223` LEDVANCE SMART+ Filament Edison 52 E27 Amber dimmable
- `4058075729247` LEDVANCE SMART+ Filament Globe125 52 E27 Amber dimmable
- `4058075729308` LEDVANCE SMART+ Outdoor Plug EU
- `4058075729322` LEDVANCE SMART+ Compact Outdoor Plug EU
- `40ZH-O` TuYa Motion sensor
- `43113` Enbrighten Zigbee in-wall smart dimmer
- `5309031P8` Philips Hue White ambiance Runner spot white (1 spot)
- `552-80698` Niko Smart plug with side earthing pin
- `8718696166079` Philips Hue Tuar outdoor wall light
- `8719514279193` Philips Hue white filament Edison ST72 B22 LED
- `915005997901` Philips Hue White & Color Ambiance Xamento L
- `915005998001` Philips Hue white ambiance pendant white Enrave
- `915005998101` Philips Hue white ambiance pendant black Enrave
- `929003045901` Philips Hue White ambiance Runner spot white (1 spot)
- `929003052601` Philips Hue Flourish white and color ambiance table light with Bluetooth
- `929003052901` Philips Hue Sana wall light (black)
- `9290031345` Philips Hue white and color ambiance 4" retrofit recessed downlight
- `AU-A1GSZ9CX` Aurora AOne GLS lamp 9w tunable dimmable 2200-5000K
- `AU-A1ZBSCRGBCX` Aurora RGBW LED strip controller
- `C10-3E-1.2` Novo Curtain switch
- `DDS238-2` TuYa Zigbee smart energy meter
- `GL-C-004P` Gledopto Zigbee LED Strip Light Kit
- `GW003-AS-IN-TE-FC` Atlantic Group Interface Naviclim for Takao air conditioners
- `HG08673-FR` Lidl Silvercrest smart plug FR with power monitoring
- `HS2WDS` HEIMAN LED 9W CCT E27
- `HS2WDSC-E` HEIMAN Remote dimmer and temperature control
- `HS2WDSR-E` HEIMAN Remote dimmer and color control
- `LA-PLUG-10Amp` Ynoa Smart plug Zigbee 3.0
- `LED2009C3` IKEA TRADFRI LED bulb E12 WW candle clear 250 lumen, dimmable
- `LED2101G4` IKEA TRADFRI bulb E14 WS globe 470lm, dimmable, white spectrum, opal white
- `PCT504` OWON HVAC fan coil
- `SA12IZL` TuYa Smart smoke alarm
- `SNZB-02_EFEKTA` Custom devices (DiY) Alternative firmware for the SONOFF SNZB-02 sensor from EfektaLab, DIY
- `SRTS-A01` Xiaomi Aqara Smart Radiator Thermostat E1
- `SSWF01G` Mercator AC fan controller
- `T2035` IKEA STOFTMOLN ceiling/wall lamp 24 warm light dimmable
- `T2037` IKEA STOFTMOLN ceiling/wall lamp 37 warm light dimmable
- `TS0001_power` TuYa Switch with power monitoring
- `Touch4` databyte.ch Wall touchsensor with 4 keys
- `WG001` Aeotec Range extender Zi
- `ZBDongle-E` SONOFF Sonoff Zigbee 3.0 USB Dongle Plus (EFR32MG21) with router firmware
- `ZM25RX-08/30` Zemismart Tubular motor
- `ZS-EUB_1gang` Moes Wall light switch (1 gang)
- `ZS057-D0Z` Leviton Wall switch, 0-10V dimmer, 120-277V, Lumina™ RF
- `ZYA-C4-MOD-S` Yale Control4 module for Yale KeyFree/Keyless/Doorman/Assure/nexTouch locks

<hr /><em>This discussion was created from the release <a href='https://github.com/Koenkk/zigbee2mqtt/releases/tag/1.28.0'>1.28.0</a>.</em>

**Comment by Uniranger** [comment]
您好,我已收到您的来信,我会尽快查看,谢谢。
